src/components/dateTimePicker/index.js

```diff
@@ -144,20 +148,20 @@ class DateTimePicker extends BaseComponent {
 
   getStringValue = () => {
     const {value} = this.state;
-    const {mode, dateFormat, timeFormat, dateFormatter} = this.getThemeProps();
+    const {mode, dateFormat, timeFormat, dateFormatter, timeFormatter} = this.getThemeProps();
     if (value) {
-      if (dateFormatter) {
-        return dateFormatter(value);
+      switch (mode) {
+        case MODES.DATE:
+          if (dateFormatter) {
+            return dateFormatter(value);
+          }
+          return dateFormat ? moment(value).format(dateFormat) : value.toLocaleDateString();
```

Comment:
You can have all in one trinery: 
return dateFormatter ?  dateFormatter(value) : dateFormat ? moment(value).format(dateFormat) : value.toLocaleDateString();